eo-maven-plugin/src/main/java/org/eolang/maven/fp/Cache.java

```diff
@@ -0,0 +1,140 @@
+ * SOFTWARE.
+ */
+package org.eolang.maven.fp;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Cache directory.
+ * @since 0.41.0
+ */
+public final class Cache {
+    /**
+     * Base cache directory.
+     */
+    private final Path base;
+
+    /**
+     * Cache version.
+     */
+    private final Cache.Version version;
+
+    /**
+     * Relative path of base and version.
+     */
+    private final Path relative;
+
+    /**
+     * Ctor.
+     * @param base Base cache directory
+     * @param version Cache version
+     * @param relative Relative path
+     */
+    public Cache(final Path base, final Cache.Version version, final Path relative) {
+        this.base = base;
+        this.version = version;
+        this.relative = relative;
+    }
+
+    /**
+     * Checks if the current cache directory is cacheable.
+     * The directory is cacheable if cache version is cacheable.
+     * @return True if cacheable
+     */
+    public boolean cacheable() {
+        return this.version.cacheable();
+    }
+
+    /**
+     * Return full absolute path to cache directory.
+     * @return Path to cache directory
+     */
+    public Path path() {
+        return this.base.resolve(this.version.path()).resolve(this.relative);
+    }
+
+    /**
+     * Cache version.
+     * The version consists of two main components:
+     *  1) Base, which is maven-style version string
+     *  2) Hash string, containing git hash.
+     * The two parts define the cacheable location by method {@link #path}.
+     * If base is empty or contains '0.0.0' or 'SNAPSHOT' (see {@link #NOT_CACHEABLE}
+     * versions) the version is considered non-cacheable.
+     * @since 0.41.0
+     */
+    public static class Version {
+        /**
+         * Not cacheable versions.
+         */
+        private static final String[] NOT_CACHEABLE = {"0.0.0", "SNAPSHOT"};
+
+        /**
+         * Version of the eo-maven-plugin which currently builds a program. Version could be:
+         * - 0.0.0 - if the version is not set in the pom.xml
+         * - 0.30.0 - if the version is set in the pom.xml
+         * - 1.0-SNAPSHOT - if the not stable development version is set in the pom.xml By some
+         * reason, the version could also be empty.
+         */
+        private final String semver;
+
+        /**
+         * Hash of tag from the objectionary/home.
+         */
+        private final String hash;
+
+        /**
+         * Ctor.
+         * @param ver Version of the eo-maven-plugin which currently builds a program.
+         * @param hsh Hash of the objectionary tag.
+         */
+        public Version(final String ver, final String hsh) {
+            this.semver = ver;
+            this.hash = hsh;
+        }
+
+        /**
+         * Get cache path.
+         * @return Path.
+         */
+        Path path() {
+            return Paths.get(Objects.toString(this.semver, ""))
```

Comment:
@maxonfjvipon what's the purpose of `Objects.toString` here? The `semver` is a string already. Or you are trying to avoid NPE here? Maybe it's better to not allow NULLs in the constructor?